src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs

```diff
@@ -104,18 +114,19 @@ public async Task StartAsync(CancellationToken cancellationToken)
 
             if (isPastDue)
             {
-                _logger.LogDebug($"Function '{_timerName}' is past due on startup. Executing now.");
-                await InvokeJobFunction(now, isPastDue: true);
+                _logger.LogDebug($"Function '{_functionShortName}' is past due on startup. Executing now.");
+                await InvokeJobFunction(now, isPastDue: true, originalSchedule: ScheduleStatus.Next);
             }
             else if (_attribute.RunOnStartup)
             {
                 // The job is configured to run immediately on startup
-                _logger.LogDebug($"Function '{_timerName}' is configured to run on startup. Executing now.");
+                _logger.LogDebug($"Function '{_functionShortName}' is configured to run on startup. Executing now.");
+                _logger.LogWarning($"Function '{_functionShortName}' has '{nameof(TimerTriggerAttribute.RunOnStartup)}' set to 'true'. This property is intended for testing purposes only and may result in unintended behavior if used in production.");
```

Comment:
I don't think we should add this one. For usages outside of Functions (long running continuous WebJobs) this isn't true and using RunOnStartup is perfectly reasonable. It's only in the context of Azure Functions and short lived host instances that people confuse themselves with this.

Reply:
Fair enough -- I figured there was a reason we hadn't added it already. I had a CRI on this last week (and @mhoeger suggested it as well) but I wasn't thinking of WebJobs. The TriggerDetails noting that the invocation happened b/c of RunOnStartup should help people identify this themselves (I hope).

Reply:
@brettsam - we'd have to do some hard-coding, but we could also check for that log in a detector. Any chance we could configure that message to be a little more query-friendly?

Reply:
This warning is now gone -- but we still log the details if it's been run b/c of RunOnStartup.

The log looks like: `Trigger Details: UnscheduledInvocationReason: RunOnStartup`. So we should look for `UnscheduledInvocationReason` -- that should also have the FunctionName and FunctionInvocationId associated with it.